tianshou/data/buffer.py

```diff
@@ -520,13 +437,215 @@ def update_weight(
     def __getitem__(
         self, index: Union[slice, int, np.integer, np.ndarray]
     ) -> Batch:
-        return Batch(
-            obs=self.get(index, "obs"),
-            act=self.act[index],
-            rew=self.rew[index],
-            done=self.done[index],
-            obs_next=self.get(index, "obs_next"),
-            info=self.get(index, "info"),
-            policy=self.get(index, "policy"),
-            weight=self.weight[index],
-        )
+        batch = super().__getitem__(index)
+        batch.weight = self.weight[index]
+        return batch
+
+
+class ReplayBuffers(ReplayBuffer):
+    """ReplayBuffers contains a list of ReplayBuffer.
+
+    These replay buffers have contiguous memory layout, and the storage space
+    each buffer has is a shallow copy of the topmost memory.
+
+    :param int buffer_list: a list of ReplayBuffers needed to be handled.
+
+    .. seealso::
+
+        Please refer to :class:`~tianshou.data.ReplayBuffer` for more detailed
+        explanation.
+    """
+
+    def __init__(self, buffer_list: List[ReplayBuffer], **kwargs: Any) -> None:
+        self.buffer_num = len(buffer_list)
```

Comment:
I think we need to check if buffer_list meta is legal(empty), otherwise it will cause error

Reply:
Also, we should make sure data type is the most basic ReplayBuffer